### PR TITLE
Display the workflow oauth scope for GHES 3.0 and above, GHAE, and GHEC

### PR DESCRIPTION
**Pull request by niik**
### Why:

I'm working on having GitHub Desktop request the `workflow` scope on GitHub Enterprise Server and was surprised when I couldn't find the it in the documentation. The `workflow` scope was officially added in [GitHub Enterprise 3.0](https://github.blog/2021-02-16-github-enterprise-server-3-0-is-now-generally-available/) alongside the GA of GitHub Actions.

I've confirmed this by launching a test instance of GHES 3.0.

### What's being changed:

The [Scopes for OAuth Apps](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps) documentation page documents the `workflow` scope for "Free, Pro & Team" and "Enterprise Cloud" but not for any GitHub Enterprise version even though it's been available since 3.0.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@niik 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by lecoursen — APPROVED** [review]
Looks great, I'll merge this now! ⚡

**Review by Pojielp — COMMENTED** [review]
4225fdc066752eaee3ea696c102a787705d74789

File: content/developers/apps/building-oauth-apps/scopes-for-oauth-apps.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ Name | Description
 **`admin:gpg_key`** | Fully manage GPG keys.
 &emsp;`write:gpg_key`| Create, list, and view details for GPG keys.
 &emsp;`read:gpg_key`| List and view details for GPG keys.{% ifversion fpt or ghec %}
-**`codespace`** | Grants the ability to create and manage codespaces. Codespaces can expose a GITHUB_TOKEN which may have a different set of scopes. For more information, see "[Security in Codespaces](/codespaces/codespaces-reference/security-in-codespaces#authentication)."
```
Comment (AlexanderCsr8904) [inline content/developers/apps/building-oauth-apps/scopes-for-oauth-apps.md:77]: #13918 Searching

**Pull request merged**